### ISSUE TITLE
no-static-element-interactions shows error when "role" is applied as a variable

### ISSUE THREAD

**Issue by Nick-Mazuk** [jsx-eslint/eslint-plugin-jsx-a11y#745]
Consider the following example:

```tsx
const role = 'button'

return (
    <div
        onClick={handleClick}
        role={role}
        tabIndex={0}
    >
        {children}
    </div>
)
```

This code results in the following error:

```
Static HTML elements with event handlers require a role. eslint(jsx-a11y/no-static-element-interactions)
```

even though the div element has a role. It's just applied dynamically.

**Expected behavior**

The above code snippet would not result in an error because it is equivalent to the following snippet which does not produce an error:

```tsx
return (
    <div
        onClick={handleClick}
        role='button'
        tabIndex={0}
    >
        {children}
    </div>
)
```

**Comment by ljharb** [comment]
For static analysis tools to work, things have to be static. Applying the role dynamically defeats this.

In the trivial case where it's a string literal that's never reassigned and is defined in the same file, the rule could be extended to support it - but I'm not sure what the value is to support this pattern.

**Comment by Nick-Mazuk** [comment]
> For static analysis tools to work, things have to be static. Applying the role dynamically defeats this.
> 
> In the trivial case where it's a string literal that's never reassigned and is defined in the same file, the rule could be extended to support it - but I'm not sure what the value is to support this pattern, especially when it would be an extreme minority of the non-inline cases.

I see. Because I use this specific pattern for clickable elements a lot, I was building an HOC to add all the details, with the `role` attribute being an optional prop.

But I totally see your point.
